relay-profiling/src/android.rs

```diff
@@ -141,83 +133,23 @@ pub fn expand_android_profile(payload: &[u8]) -> Result<Vec<Vec<u8>>, ProfileErr
         return Ok(items);
     }
 
-    for transaction in &profile.transactions {
-        let mut new_profile = profile.clone();
-
-        new_profile.profile_id = EventId::new();
-        new_profile.set_transaction(transaction);
-        new_profile.transactions.clear();
-
-        let clock = new_profile.profile.clock;
-        let start_time = new_profile.profile.start_time;
-
-        new_profile.profile.events.retain_mut(|event| {
-            let event_timestamp = get_timestamp(clock, start_time, event.time);
-            if transaction.relative_start_ns <= event_timestamp
-                && event_timestamp <= transaction.relative_end_ns
-            {
-                let relative_start = match clock {
-                    Clock::Cpu => Duration::from_millis(transaction.relative_cpu_start_ms),
-                    Clock::Wall | Clock::Dual | Clock::Global => {
-                        Duration::from_nanos(transaction.relative_start_ns)
-                    }
-                };
-                event.time = substract_to_timestamp(event.time, relative_start);
-                true
-            } else {
-                false
-            }
-        });
+    let transaction = &profile.transactions[0];
```

Comment:
This can panic if `profile.transactions` is empty but `has_transaction_metadata` returns false.

Reply:
Fixed in 0c109eb324e50c7e087ce9d1365a2046a76fd535. @phacops could you double-check that the logic was even correct? It seems to me as if the _intent_ of this was rather to check `if transactions.is_empty() || profile.has_tx() { return }`

Reply:
The logic was correct. This was to provide compatibility with the previous format where we'd receive transaction metadata as top level fields. If we had an empty `transactions` array and top level fields (as previous Android SDKs were sending the profile), we can just encode the payload and move on.